Content.Server/Forensics/Systems/ForensicsSystem.cs

```diff
@@ -101,17 +105,61 @@ public void CopyForensicsFrom(ForensicsComponent src, EntityUid target)
             }
         }
 
-        private void OnAfterInteract(EntityUid uid, CleansForensicsComponent component, AfterInteractEvent args)
+        private void OnAfterInteract(Entity<CleansForensicsComponent> cleanForensicsEntity, ref AfterInteractEvent args)
         {
             if (args.Handled || !args.CanReach)
                 return;
 
-            if (!TryComp<ForensicsComponent>(args.Target, out var forensicsComp))
+            args.Handled = TryStartCleaning(cleanForensicsEntity, args.User, args.Target);
+        }
+
+        private void OnUtilityVerb(Entity<CleansForensicsComponent> entity, ref GetVerbsEvent<UtilityVerb> args)
+        {
+            if (!args.CanInteract || !args.CanAccess)
                 return;
 
-            if((forensicsComp.DNAs.Count > 0 && forensicsComp.CanDnaBeCleaned) || (forensicsComp.Fingerprints.Count + forensicsComp.Fibers.Count > 0))
+            // These need to be set outside for the anonymous method!
+            var user = args.User;
+            var target = args.Target;
+
+            var verb = new UtilityVerb()
+            {
+                Act = () =>
+                {
+                    if (!TryStartCleaning(entity, user, target))
+                    {
+                        _popupSystem.PopupEntity(Loc.GetString("forensics-cleaning-cannot-clean", ("target", target)), user, user, PopupType.MediumCaution);
+                    }
```

Comment:
this would be fine if the method was private, but since its public behavior like failure popups should be whiting the method even if it leads to a bit more redundant code

Reply:
I don't quite get what your saying! What does "whiting the method" mean (Sorry if this is a silly question!)?

Reply:
i meant instead of putting the popup feedback outside of the method when it fails, put it inside the method when it fails

Reply:
That makes sense! Now I see what you meant by redundant code. Let me know if that looks good!